src/main/scala/com/fulcrumgenomics/umi/ReviewConsensusVariants.scala

```diff
@@ -52,82 +129,197 @@ import scala.collection.mutable.ListBuffer
+      require(variant.start == consensusPileup.getPosition,     "Variants and consensus iterator out of sync.")
+      require(variant.start == groupedPileup.getPosition,       "Variants and grouped iterator out of sync.")
+
+      // Setup ready to iterate through the consensus reads
+      val consensusCounts  = BaseCounts(consensusPileup.getRecordAndOffsets.asScala)
+
+      val rawByMiAndReadNum = groupedPileup.getRecordAndOffsets.asScala.groupBy(r => toMi(r.getRecord) + readNumberSuffix(r.getRecord))
+
+      consensusPileup.getRecordAndOffsets
+        .filter(r => r.getReadBase.toChar.toUpper != variant.refBase)
+        .filterNot(r => this.ignoreNsInConsensusReads && r.getReadBase.toChar.toUpper == 'N')
+        .toSeq.sortBy(r => toMi(r.getRecord) + readNumberSuffix(r.getRecord))
+        .foreach { c =>
+          val mi = toMi(c.getRecord)
+          val consensusReadName = c.getRecord.getReadName + ( if (c.getRecord.getReadPairedFlag && c.getRecord.getSecondOfPairFlag) "/2" else "/1" )
+          val rawCounts = BaseCounts(rawByMiAndReadNum(mi + readNumberSuffix(c.getRecord)))
+
+          val m =  ConsensusVariantReviewInfo(
+            chrom = variant.chrom,
+            pos   = variant.start,
+            ref   = variant.refBase.toString,
+            genotype = variant.genotype.getOrElse("NA"),
+            filters  = variant.filters.getOrElse(""),
+            A = consensusCounts.a,
+            C = consensusCounts.c,
+            G = consensusCounts.g,
+            T = consensusCounts.t,
+            N = consensusCounts.n,
+            consensus_read = consensusReadName,
+            consensus_insert = toInsertString(c.getRecord),
+            consensus_call = c.getReadBase.toString.toUpperCase,
+            consensus_qual = c.getBaseQuality,
+            a = rawCounts.a,
+            c = rawCounts.c,
+            g = rawCounts.g,
+            t = rawCounts.t,
+            n = rawCounts.n
+          )
+
+          writer.write(m)
+        }
+    }
+
+    writer.close()
   }
 
   /** Loads the set of locations for review. */
-  private[umi] def loadPositions(path: FilePath, refFile: ReferenceSequenceFile): Seq[LocAndRef] = {
-    val buffer = ListBuffer[LocAndRef]()
+  private[umi] def loadPositions(path: FilePath, refFile: ReferenceSequenceFile): Seq[Variant] = {
+    val buffer = ListBuffer[Variant]()
 
     if (VcfExtensions.exists(ext => path.getFileName.toString.endsWith(ext))) {
       val in = new VCFFileReader(input.toFile, false)
-      in.filter(_.isSNP).foreach(v => buffer += LocAndRef(v.getContig, v.getStart, v.getReference.getBases()(0).toChar.toUpper))
+      val sample = this.sample.orElse {
+        if (in.getFileHeader.getSampleNamesInOrder.size() == 1) Some(in.getFileHeader.getSampleNamesInOrder.get(0))
+        else None
+      }
+
+      in.filter(_.isSNP).filter(v => mafFromGenotype(v, sample).forall(_ <= this.maf)).foreach { v => buffer += Variant(
```

Comment:
`isSNP` is for single-nucleotide variants, though it can include ones with > 1 alt allele.  However, we're looking at a single sample's genotype which I'm making the assumption is diploid.